Applications/Utils/ModelPreparation/PartitionMesh/MeshPartitioning.cpp

```diff
@@ -0,0 +1,493 @@
+/*!
+  \file MeshPartitioning.cpp
+  \date   2016.05
+
+  \brief  Define the members of class MeshPartitioning
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#include "MeshPartitioning.h"
+
+#include <iomanip>
+#include <stdio.h> // for binary output
```

Comment:
This C-io is only for the binary output, which  is for MPI parallel reading by MPI_File_read. Since MPI_File_read  is based on printf function, printf is used in this file for binary output only. 

I have ever tried to use  stream.write to dump the binary file, but did not succeed in making it read by  MPI_File_read.